images/secret-sync/secret-sync.py

```diff
@@ -0,0 +1,162 @@
+    except:
+        config.load_incluster_config()
+
+    v1 = client.CoreV1Api()
+    try:
+        secret = v1.read_namespaced_secret(namespace=namespace, name=secret_name)
+    except client.rest.ApiException as e:
+        if e.status == 404:
+            secret = client.V1Secret(
+                metadata=client.V1ObjectMeta(name=secret_name, labels=labels),
+                data={}
+            )
+            resp = v1.create_namespaced_secret(namespace=namespace, body=secret)
+            logging.info(f"Created secret {secret_name} since it does not exist")
+        else:
+            raise
+    # Value should be base64'd string
+    new_value = base64.standard_b64encode(value).decode()
+    if secret.data is None:
+        secret.data = {}
+    if new_value != secret.data.get(key):
+        secret.data[key] = base64.standard_b64encode(value).decode()
+        v1.patch_namespaced_secret(namespace=namespace, name=secret_name, body=secret)
+        logging.info(f"Updated secret {secret_name} with new value for key {key}")
+
+def get_secret_value(namespace, secret_name, key):
+    try:
+        config.load_kube_config()
+    except:
+        config.load_incluster_config()
+
+    v1 = client.CoreV1Api()
+    try:
+        secret = v1.read_namespaced_secret(namespace=namespace, name=secret_name)
+    except client.rest.ApiException as e:
+        if e.status == 404:
+            # Secret doesn't exist
+            return None
+        raise
+    if secret.data is None or key not in secret.data:
+        return None
+    return base64.standard_b64decode(secret.data[key])
+
+def setup_logging():
+    """
+    Set up root logger to log to stderr
+    """
+    logging.basicConfig(format="%(asctime)s %(message)s", level=logging.INFO, stream=sys.stderr)
+
+def main():
+    argparser = argparse.ArgumentParser()
+    argparser.add_argument(
+        '--namespace',
+        help='Namespace the secret exists in'
+    )
+
+    argparser.add_argument(
+        'action',
+        choices=['load', 'watch-save']
+    )
+
+    argparser.add_argument(
+        'secret_name',
+        help="Name of secret to sync with. Will be created if needed."
+    )
+
+    argparser.add_argument(
+        'key',
+        help="Key in secret object to sync file to"
+    )
+
+    argparser.add_argument(
+        'path',
+        help="Path in filesystem to sync to"
+    )
+
+    argparser.add_argument(
+        '--label',
+        help="Labels (of form key=value) to add to the k8s secret when it is created",
+        action="append"
+    )
+
+    args = argparser.parse_args()
+
+    setup_logging()
+
+    if not args.namespace:
+        try:
+            with open("/var/run/secrets/kubernetes.io/serviceaccount/namespace") as f:
+                args.namespace = f.read().strip()
+        except FileNotFoundError:
+            print("Can not determine a namespace, must be explicitly set with --namespace", file=sys.stderr)
+            sys.exit(1)
```

Comment:
Oh, I'm learning seeing this. Looked up stuff in the docs found by googling /var/run/secrets/kubernetes.io

https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#service-account-admission-controller